pkg/apis/build/v1alpha2/build_pod.go

```diff
@@ -1194,3 +1221,12 @@ func envs(envs []corev1.EnvVar, envVar corev1.EnvVar) []corev1.EnvVar {
 	}
 	return envs
 }
+
+// Volume names must be valid RFC 1123 Label Names: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names
+func secretNameToVolumeName(secretName string) (string, error) {
+	volumeName := fmt.Sprintf(volumeSecretNameTemplate, secretName)
+	if len(volumeName) > 63 {
+		return "", errors.Errorf("secret name %s is too long, it must contain fewer than 50 characters", secretName)
+	}
+	return strings.ReplaceAll(volumeName, ".", "-"), nil
```

Comment:
[This is reaaaally theoretical but, channeling my inner](https://github.com/pivotal/kpack/pull/947#discussion_r846638162) @tomkennedy513 is there a risk for collision here? 

`my.secret` and `my-secret` will have the same volume name?

Reply:
added uniqueness!